src/Powercord/plugins/pc-announcements/index.js

```diff
@@ -73,16 +79,17 @@ module.exports = class Announcements extends Plugin {
   }
 
   _renderNotice () {
-    if (document.querySelector('.pc-guildsWrapper + .pc-flex > .pc-flexChild .pc-notice')) {
+    const Notice = require('./Notice');
```

Comment:
why put the require here?

Reply:
> why put the require here?

To answer **why** I've chosen to move the require here compared to on `Line 10` is mainly due to the fact that the `Notice.TYPES` store is called too early (way before the notice classes from `getModule([ 'noticeBrand' ])` are even initialized).

That said, with the constant variable staying in it's original spot (on line 10) you'll find that the classes for announcements/notices can't be retrieved, causing them to be `undefined` and ultimately preventing styling (CSS) to both the `<div>` and its corresponding elements (`<button>` and `<Clickable>`).

Reply:
I see why the require was moved, but since _renderNotice can still be called early it can still cause race conditions in case of slow boots. In cases where the module is not initialized, Powercord will return a promise and it breaks as it's not what you'd expect.

A workarround can be to not pass classes directly in Notice.TYPES but just give constants (but keep the object for autocomplete/compatibility purposes) (`BLURPLE: 'BLURPLE'`) and mount classes in the `componentDidMount` with await and all promise stuff. Just use setState to store classes and make the component re-render, and boom.

What you can also do is save it in a variable and initialize store with it immediately, so if the module already have been fetched, it won't fetch it again.

Reply:
Marking this conversation as **Resolved** - workaround can be found in the latest commit. Let me know if anything else needs to be changed.